### PR TITLE
[aot_autograd] symintify `logsumexp`

### PR DESCRIPTION
**Pull request by nkaretnikov**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #91442

### CODE REVIEW

**Comment by nkaretnikov** [comment]
Also fails inductor tests. I've spent a bit of time trying to fix those as well, but it's not clear how much work is required. So only submitting this for now.

File: torch/csrc/autograd/FunctionsManual.cpp [hunk 1]
```diff
@@ -791,8 +791,8 @@ Tensor logsumexp_backward(
     IntArrayRef dim,
     bool keepdim) {
   if (!keepdim && self.dim() != 0) {
-    grad = unsqueeze_multiple(grad, dim, self.sizes().size());
-    result = unsqueeze_multiple(result, dim, self.sizes().size());
+    grad = unsqueeze_multiple(grad, dim, self.sym_sizes().size());
+    result = unsqueeze_multiple(result, dim, self.sym_sizes().size());
```
Comment (albanD) [inline torch/csrc/autograd/FunctionsManual.cpp:795]: You can replace these with `self.dim()` as well :)

**Review by albanD — APPROVED** [review]
Small improvement but LGTM in any case.

**Pull request closed without merging**